setup.go

```diff
@@ -0,0 +1,822 @@
+}
+
+func (opts *setupOptionsType) askInventoryPoll(ctx *cli.Context,
+	stdin *stdinReader) error {
+	if !ctx.IsSet("inventory-poll") ||
+		opts.invPollInterval < minimumPollInterval {
+		rsp, err := stdin.promptUser(
+			promptInventoryPoll, false)
+		if err != nil {
+			return err
+		}
+		for {
+			if rsp == "" {
+				opts.invPollInterval = defaultInventoryPoll
+				break
+			} else if opts.invPollInterval, err = strconv.Atoi(
+				rsp); err != nil {
+				rsp, err = stdin.promptUser(
+					rspNotSeconds, false)
+			} else if opts.invPollInterval < minimumPollInterval {
+				rsp, err = stdin.promptUser(
+					rspInvalidInterval, false)
+			} else {
+				break
+			}
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (opts *setupOptionsType) askRetryPoll(ctx *cli.Context,
+	stdin *stdinReader) error {
+	if !ctx.IsSet("retry-poll") ||
+		opts.retryPollInterval < minimumPollInterval {
+		rsp, err := stdin.promptUser(
+			promptRetryPoll, false)
+		if err != nil {
+			return err
+		}
+		for {
+			if rsp == "" {
+				opts.retryPollInterval = defaultRetryPoll
+				break
+			} else if opts.retryPollInterval, err = strconv.Atoi(
+				rsp); err != nil {
+				rsp, err = stdin.promptUser(
+					rspNotSeconds, false)
+			} else if opts.retryPollInterval < minimumPollInterval {
+				rsp, err = stdin.promptUser(
+					rspInvalidInterval, false)
+			} else {
+				break
+			}
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (opts *setupOptionsType) askPollingIntervals(ctx *cli.Context,
+	stdin *stdinReader) (int, error) {
+	if err := opts.askUpdatePoll(ctx, stdin); err != nil {
+		return -1, err
+	}
+	if err := opts.askInventoryPoll(ctx, stdin); err != nil {
+		return -1, err
+	}
+	if err := opts.askRetryPoll(ctx, stdin); err != nil {
+		return -1, err
+	}
+	return stateDone, nil
+}
+
+func doSetup(ctx *cli.Context, config *menderSysConfig,
+	opts *setupOptionsType) error {
+	var err error
+	state := stateDeviceType
+	stdin := &stdinReader{
+		reader: bufio.NewReader(os.Stdin),
+	}
+
+	// Prompt 'wizard' message
+	fmt.Println(promptWizard)
+
+	// Prompt the user for config options if not specified by flags
+	for state != stateDone {
+		switch state {
+		case stateDeviceType:
+			state, err = opts.askDeviceType(ctx, stdin)
+
+		case stateDeviceTypeConfirm:
+			state, err = opts.askConfirmDeviceType(ctx, stdin)
+
+		case stateHostedMender:
+			state, err = opts.askHostedMender(ctx, stdin)
+
+		case stateDemoMode:
+			state, err = opts.askDemoMode(ctx, stdin)
+
+		case stateServerURL:
+			state, err = opts.askServerURL(ctx, stdin)
+
+		case stateServerIP:
+			state, err = opts.askServerIP(ctx, stdin)
+
+		case stateServerCert:
+			state, err = opts.askServerCert(ctx, stdin)
+
+		case stateHostedMenderCredentials:
+			state, err = opts.askHostedMenderCredentials(ctx, stdin)
+
+		case statePolling:
+			state, err = opts.askPollingIntervals(ctx, stdin)
+		}
+		if err != nil {
+			return err
+		}
+	} // END for {state}
```

Comment:
We need some simple message here that says to the user that the configuration is done. As it is now, it exists silently unless you have passed the `--run-daemon` flag